jobs/furnishings/furnish.py

```diff
@@ -65,11 +76,190 @@ def shell_context():
+        email: str = None
+        ):
+    """Create new furnishing entry."""
+    business = batch_processing.business
+    if business.legal_type == Business.LegalTypes.EXTRA_PRO_A.value:
+        furnishing_name = (
+            Furnishing.FurnishingName.DISSOLUTION_COMMENCEMENT_NO_TR_XPRO
+            if eligible_details.transition_overdue
+            else Furnishing.FurnishingName.DISSOLUTION_COMMENCEMENT_NO_AR_XPRO
+        )
+    else:
+        furnishing_name = (
+            Furnishing.FurnishingName.DISSOLUTION_COMMENCEMENT_NO_TR
+            if eligible_details.transition_overdue
+            else Furnishing.FurnishingName.DISSOLUTION_COMMENCEMENT_NO_AR
+        )
+
+    new_furnishing = Furnishing(
+        furnishing_type=furnishing_type,
+        furnishing_name=furnishing_name,
+        batch_id=batch_processing.batch_id,
+        business_id=batch_processing.business_id,
+        business_identifier=batch_processing.business_identifier,
+        processed_date=datetime.utcnow(),
+        created_date=datetime.utcnow(),
+        last_modified=datetime.utcnow(),
+        status=Furnishing.FurnishingStatus.QUEUED,
+        email=email
+    )
+    new_furnishing.save()
+
+    return new_furnishing
+
+
+async def stage_1_process(app: Flask, qsm: QueueService):  # pylint: disable=redefined-outer-name
+    """Run process to manage and track notifications for dissolution stage 1 process."""
+    try:
+        batch_processings = (
+            db.session.query(BatchProcessing)
+            .filter(BatchProcessing.status == BatchProcessing.BatchProcessingStatus.PROCESSING)
+            .filter(BatchProcessing.step == BatchProcessing.BatchProcessingStep.WARNING_LEVEL_1)
+            .filter(Batch.id == BatchProcessing.batch_id)
+            .filter(Batch.batch_type == Batch.BatchType.INVOLUNTARY_DISSOLUTION)
+            .filter(Batch.status == Batch.BatchStatus.PROCESSING)
+        ).all()
+
+        for batch_processing in batch_processings:
+            furnishings = Furnishing.find_by(
+                batch_id=batch_processing.batch_id,
+                business_id=batch_processing.business_id
+                )
+            if not furnishings:
+                # send email/letter notification for the first time
+                email = get_email_address_from_auth(batch_processing.business_identifier)
+                if email:
+                    # send email letter
+                    _, eligible_details = InvoluntaryDissolutionService.check_business_eligibility(
+                        batch_processing.business_identifier, False
+                        )
+                    if eligible_details:
+                        new_furnishing = create_new_furnishing(
+                            batch_processing,
+                            eligible_details,
+                            Furnishing.FurnishingType.EMAIL,
+                            email
+                            )
```

Comment:
I thought about the design a bit more and I think for every all furnishing entries of a given furnishing type should have a `grouping_identifier` assigned.  

Can you generate one and pass it into each `create_new_furnishing` call here?